Guard against negative rpcTimeout being passed in to RpcBackendOptions

osalpekar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#38267 Guard against negative rpcTimeout being passed in to RpcBackendOptions**

Assert that the rpcTimeout is positive in RpcBackendOptions
constructor

Differential Revision: [D21509850](https://our.internmc.facebook.com/intern/diff/D21509850/)

osalpekar committed:
Guard against negative rpcTimeout being passed in to RpcBackendOptions

Assert that the rpcTimeout is non-negative in RpcBackendOptions
constructor

Differential Revision: [D21509850](https://our.internmc.facebook.com/intern/diff/D21509850/)

[ghstack-poisoned]

osalpekar committed:
Update on "Guard against negative rpcTimeout being passed in to RpcBackendOptions"


Assert that the rpcTimeout is positive in RpcBackendOptions
constructor

Differential Revision: [D21509850](https://our.internmc.facebook.com/intern/diff/D21509850/)

[ghstack-poisoned]

mrshenli commented on torch/csrc/distributed/rpc/rpc_agent.h:
```diff
@@ -32,7 +32,9 @@ struct RpcBackendOptions {
       : rpcTimeoutSeconds(rpcTimeoutSeconds),
-        initMethod(std::move(initMethod)) {}
+        initMethod(std::move(initMethod)) {
+    TORCH_CHECK(rpcTimeoutSeconds > 0, "RPC Timeout must be greater than 0");
```
Do we accept 0 timeout as infinite timeout? The args doc in `rpc_sync` says:

```
        timeout (float, optional): timeout in seconds to use for this RPC. If
                                   the RPC does not complete in this amount of
                                   time, an exception indicating it has
                                   timed out will be raised. A value of 0
                                   indicates an infinite timeout, i.e. a timeout
                                   error will never be raised. If not provided,
                                   the default value set during initialization
                                   or with `_set_rpc_timeout` is used.
```

osalpekar replied:
Yes, looks like we should support 0 - will update

mrshenli commented:
Flake8 error has been fixed on master. A rebase should help get rid of the lint error.

osalpekar committed:
Update on "Guard against negative rpcTimeout being passed in to RpcBackendOptions"


Assert that the rpcTimeout is positive in RpcBackendOptions
constructor

Differential Revision: [D21509850](https://our.internmc.facebook.com/intern/diff/D21509850/)

[ghstack-poisoned]
